workspace/template.ts

```diff
@@ -1,7 +1,73 @@
+
+const readline = require('readline');
+const request = require('request');
+
+import * as _ from "lodash"
+import {Bus} from './bus'
+import {Coordinates} from './coordinates'
+import {PostCode} from './postCode'
+import {BusStop} from './busStop'
+import {Result} from './result'
+
 export class Template {
+    
+public handleUrlInput(urlPostcodeString: string):Promise<Result[]>{
+    var postCodeObject = new PostCode(urlPostcodeString);
+    let promiseCoordinates = postCodeObject.getCoordinates();
+    return promiseCoordinates.then((coords) => {
```

Comment:
As a heads up, you should be giving all of these explicit types (e.g. `...then((coords: Coordinates) => {...`), since it makes the code a lot safer.